openedx/core/lib/block_structure/block_structure.py

```diff
@@ -564,7 +564,7 @@ def remove_transformer_block_field(self, usage_key, transformer, key):
         """
         try:
             transformer_block_data = self.get_transformer_block_data(usage_key, transformer)
-            delattr(transformer_block_data, key)
+            del transformer_block_data.fields[key]
```

Comment:
@nasthagiri I wasn't able to delete the temporary field without fixing this line, is this code outdated or am I calling it wrong? The subsequent `pass`-ing `except` had been hiding the issue.

Output from my debugging session:
```
> /edx/app/edxapp/edx-platform/openedx/core/lib/block_structure/block_structure.py(567)remove_transformer_block_field()
-> delattr(transformer_block_data, key)
<openedx.core.lib.block_structure.block_structure.TransformerData object at 0x7f6152be18d0>
(Pdb) transformer_block_data.fields
{'nearest_sub_dict': (0, BlockUsageLocator(CourseLocator(u'edX', u'DemoX', u'Demo_Course', None, None), u'sequential', u'edx_introduction'))}
(Pdb) delattr(transformer_block_data, key)
*** AttributeError: 'dict' object has no attribute 'nearest_sub_dict'
(Pdb) key
'nearest_sub_dict'
(Pdb) transformer_block_data.nearest_sub_dict
(0, BlockUsageLocator(CourseLocator(u'edX', u'DemoX', u'Demo_Course', None, None), u'sequential', u'edx_introduction'))
(Pdb) del transformer_block_data.fields['nearest_sub_dict']
(Pdb) transformer_block_data.nearest_sub_dict
*** AttributeError: Field nearest_sub_dict does not exist
```